pmacApp/powerPmacAsynPortSrc/sshDriver.cpp

```diff
@@ -95,10 +103,18 @@ SSHDriver::SSHDriver(const char *host)
   // Initialize internal SSH parameters
   auth_pw_ = 0;
   got_ = 0;
-  connected_ = 0;
-  // Username and password currently set to empty strings
-  strcpy(username_, "");
-  strcpy(password_, "");
+  connected_ = SSH_NOT_CONNECTED;
+  // Initialise username and password
+  username_ = NULL;
+  password_ = NULL;
+
+  // Allocate memory required for the write method buffers
```

Comment:
Hi @ajgdls 

Would read_buffer be a more appropriate name than write_buffer, as I believe it is always used as the buffer for read data?

Also, can you give more info about the rationale behind the buffer and echo being 2 x the input.  Do they both contain the sent data and the response?

2*PPMAC_MAX_CHAR_LENGTH is used multiple times in the main code.  With this in mind, would it make sense to also make it a #define?

Reply:
Hi @JamesOHeaDLS , yes it started out as write_buffer due to its use in the write method, but it is in fact used to store bytes read from the hardware so I'll change it.
The echo buffer contains the expected response string.  If a command string contiains a CR or LF then the echo command string will contain both CR LF.  This echo string therefore has the possibility of being larger than the string it is supposedly echoing.  The factor of two is the theoretical maximum if someone sent a command which was just a lot of CR (not realistic of course).  I will create a new define for it as it is used multiple times.